pkg/secretless/plugin/sharedobj/available_plugins.go

```diff
@@ -60,35 +65,39 @@ func AllAvailablePluginsWithOptions(
 	}
 
 	httpPlugins := map[string]http.Plugin{}
+	tcpPlugins := map[string]tcp.Plugin{}
 
-	for name, httpPlugin := range internalPlugins.HTTPPlugins() {
-		if _, ok := httpPlugins[name]; ok {
-			// TODO: Should this ever happen?  Do we need this check?  Should it panic?
-			logger.Warnf("Internal plugin '%s' replaced by internal plugin", name)
+	checkNameConflicts := func(pluginType, pluginName string) {
+		// Check whether this name is a duplicate of an already-loaded HTTP plugin.
+		if _, ok := httpPlugins[pluginName]; ok {
+			logger.Panicf("%s plugin name '%s' %s", pluginType, pluginName, httpNameConflict)
+		}
+		// Check whether this name is a duplicate of an already-loaded TCP plugin.
+		if _, ok := tcpPlugins[pluginName]; ok {
+			logger.Panicf("%s plugin name '%s' %s", pluginType, pluginName, tcpNameConflict)
 		}
-		httpPlugins[name] = httpPlugin
 	}
 
-	for name, httpPlugin := range externalPlugins.HTTPPlugins() {
-		if _, ok := httpPlugins[name]; ok {
-			logger.Warnf("Internal plugin '%s' replaced by external plugin", name)
-		}
+	// Assemble internal plugins
+	for name, httpPlugin := range internalPlugins.HTTPPlugins() {
+		// Since internal plugins are hard coded, the only way that duplicate
+		// internal HTTP plugin names could occur is if the same name was
+		// erroneously used twice. So no checking for duplicate internal
+		// HTTP plugin names is done here at run time.
 		httpPlugins[name] = httpPlugin
 	}
-
-	tcpPlugins := map[string]tcp.Plugin{}
-
 	for name, tcpPlugin := range internalPlugins.TCPPlugins() {
-		if _, ok := tcpPlugins[name]; ok {
-			logger.Warnf("Internal plugin '%s' replaced by internal plugin", name)
-		}
+		checkNameConflicts("internal TCP", name)
```

Comment:
it doesn't hurt to check this, but I think the same comment as above applies
> 		// Since internal plugins are hard coded, the only way that duplicate
>		// internal plugin names could occur is if the same name was
>		// erroneously used twice. So no checking for duplicate internal
>		// plugin names is done here at run time.

Reply:
Yes, agreed. I removed the uniqueness check, and changed and moved the comment since it applies to both internal HTTP and internal TCP.